src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -263,14 +264,24 @@ object DepositHandler {
       }
     }
 
+    val itemsToResolve = fetchItems diff fetchItemsInBagStore
     for {
-      _ <- resolveFetchItems(bagitDir, fetchItems diff fetchItemsInBagStore)
+      _ <- resolveFetchItems(bagitDir, itemsToResolve)
+      _ <- removeFromFetchTxt(bagitDir, itemsToResolve)
       bag = getBagFromDir(bagitDir)
       validationResult = bag.verifyValid
       _ <- handleValidationResult(bag, validationResult, fetchItemsInBagStore)
     } yield ()
   }
 
+  private def removeFromFetchTxt(bagDir: File, items: Seq[FetchTxt.FilenameSizeUrl]): Try[Unit] = Try {
+    getFetchTxt(bagDir).foreach { case fetchTxt =>
```

Comment:
The API is a bit vague I think. Surely you cannot do both at the same time. A few tests seem to indicate that deleting file that doesn't exist of for which you have no permissions to delete it are handled by returning `false` rather than throwing an exception, so @jo-pol 's point is valid.